src/core/symbology/qgssymbollayerutils.h

```diff
@@ -897,6 +897,14 @@ class CORE_EXPORT QgsSymbolLayerUtils
      */
     static QgsStringMap evaluatePropertiesMap( const QMap<QString, QgsProperty> &propertiesMap, const QgsExpressionContext &context );
 
+    /**
+     * Calculate the minimum size in pixels of a symbol tile given the symbol \a width and \a height and the grid rotation \a angle in radians.
```

Comment:
What do you mean by "grid rotation" here? I'm concerned that this logic is ONLY used for sld export, and is a logical duplication of the code in QgsLinePatternFillSymbolLayer::applyPattern ... ultimately giving us two very complex "line pattern to image" code paths, with their own sets of unique bugs.

Reply:
Grid rotation means the rotation of the whole symbol layer (I'll change the docs to clarify), opposed to the individual marker rotation. It makes sense for point pattern fills and it is currently only used by point pattern fills, line pattern fills are simpler to calculate and do not use this method but a faster and simple formula.